src/socket/common.cr

```diff
@@ -30,6 +45,13 @@ class Socket
     INET6  = LibC::AF_INET6
   end
 
+  enum Type
```

Comment:
Is this enum used somewhere? Is there a reason why it doesn't include all the constants declared winsock?

Reply:
This is just copied over from `socket.cr` (https://github.com/crystal-lang/crystal/pull/10650/files#diff-0d9000b16ff3358d12228dd0a035fdd64fe2047fab700661a7218c034b3ca1c7L11) for the sole purpose of being able to require only `socket/common` and make `Socket::Addrinfo` compile without porting `Socket` entirely.

Reply:
Ah sorry, failed to see that.

Reply:
I probably should've split the changes into separate commits.

Reply:
I usually just read the changed files, so it would have been the same to me. I simply have to pay more attention to deletions.